File: app/src/main/java/com/tonitealive/app/ui/views/signup/SignUpFragment.java
package com.tonitealive.app.ui.views.signup;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ProgressBar;

import com.tonitealive.app.R;
import com.tonitealive.app.internal.di.components.SignUpComponent;
import com.tonitealive.app.ui.presenters.signup.SignUpPresenter;
import com.tonitealive.app.ui.views.BaseFragment;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;


public final class SignUpFragment extends BaseFragment implements SignUpView {

    public static SignUpFragment newInstance() {
        return new SignUpFragment();
    }

    @BindView(R.id.usernameField) EditText usernameField;
    @BindView(R.id.emailField) EditText emailField;
    @BindView(R.id.passwordField) EditText passwordField;
    @BindView(R.id.confirmPasswordField) EditText confirmPasswordField;
    @BindView(R.id.signUpButton) Button signUpButton;
    @BindView(R.id.progressBar) ProgressBar progressBar;

    @Inject SignUpPresenter presenter;

    @Override
    protected void initInjector() {
        SignUpComponent component = getComponentFactory().createSignUpComponent(getApplicationComponent(), this);
        component.inject(this);
    }

    @Override
    public String getUsername() {
        return usernameField.getText().toString();
    }

    @Override
    public String getEmail() {
        return emailField.getText().toString();
    }

    @Override
    public String getPassword() {
        return passwordField.getText().toString();
    }

    @Override
    public String getConfirmPassword() {
        return confirmPasswordField.getText().toString();
    }

    @Override
    public void showProgressDialog() {
        progressBar.setIndeterminate(true);
        progressBar.setVisibility(View.VISIBLE);
    }

    @Override
    public void hideProgressDialog() {
        progressBar.setVisibility(View.GONE);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_sign_up, container, false);
        ButterKnife.bind(this, view);
        return view;
    }

    @OnClick(R.id.signUpButton)
    void onSignUpButtonClicked() {
        presenter.onSignUpButtonClicked();
    }
}

File: app/src/main/kotlin/com/tonitealive/app/ui/views/signup/SignUpFragment.kt
package com.tonitealive.app.ui.views.signup

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.ProgressBar
import butterknife.BindView
import butterknife.ButterKnife
import butterknife.OnClick
import com.tonitealive.app.R
import com.tonitealive.app.ToniteAliveApplication
import com.tonitealive.app.internal.di.ComponentHolder
import com.tonitealive.app.internal.di.components.DaggerSignUpComponent
import com.tonitealive.app.internal.di.components.SignUpComponent
import com.tonitealive.app.internal.di.modules.SignUpModule
import com.tonitealive.app.ui.presenters.signup.SignUpPresenter
import com.tonitealive.app.ui.views.BaseFragment
import javax.inject.Inject


class SignUpFragment : BaseFragment(), SignUpView {

    companion object {
        fun newInstance(): SignUpFragment {
            return SignUpFragment()
        }
    }

    @BindView(R.id.usernameField) lateinit var usernameField: EditText
    @BindView(R.id.emailField) lateinit var emailField: EditText
    @BindView(R.id.passwordField) lateinit var passwordField: EditText
    @BindView(R.id.confirmPasswordField) lateinit var confirmPasswordField: EditText
    @BindView(R.id.signUpButton) lateinit var signUpButton: Button
    @BindView(R.id.progressBar) lateinit var progressBar: ProgressBar

    @Inject lateinit var presenter: SignUpPresenter

    var component by ComponentHolder<SignUpComponent>({
        val myApp = activity.application as ToniteAliveApplication
        DaggerSignUpComponent.builder()
            .applicationComponent(myApp.applicationComponent)
            .signUpModule(SignUpModule(this))
            .build()
    })

    override val username: String
        get() = usernameField.text.toString()
    override val email: String
        get() = emailField.text.toString()
    override val password: String
        get() = passwordField.text.toString()
    override val confirmPassword: String
        get() = confirmPasswordField.text.toString()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        component.inject(this)
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View {
        val view = inflater?.inflate(R.layout.fragment_sign_up, container, false)
        ButterKnife.bind(this, view!!)
        return view
    }

    override fun showProgressDialog() {
        progressBar.isIndeterminate = true
        progressBar.visibility = View.VISIBLE
    }

    override fun hideProgressDialog() {
        progressBar.visibility = View.GONE
    }

    @OnClick(R.id.signUpButton)
    fun onSignInButtonClicked() {
        presenter.onSignUpButtonClicked()
    }
}